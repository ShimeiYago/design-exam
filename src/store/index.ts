import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { loginReducer } from './login/slice';

// store
export const store = configureStore({
  reducer: {
    login: loginReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

// hooks
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
