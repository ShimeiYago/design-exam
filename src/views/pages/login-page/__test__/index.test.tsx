import { shallow, ShallowWrapper } from 'enzyme';
import { LoginPage } from '..';
import { useAppDispatch } from 'store';

jest.mock('store', () => ({
  useAppDispatch: jest.fn(),
}));

jest.mock('store/login/slice', () => ({
  login: jest.fn(),
}));

let wrapper: ShallowWrapper;

describe('Shallow Snapshot Tests', () => {
  let mockDispatch: jest.Mock;

  beforeEach(() => {
    wrapper = shallow(<LoginPage />);
    jest.clearAllMocks();
    mockDispatch = jest.fn();
    (useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
