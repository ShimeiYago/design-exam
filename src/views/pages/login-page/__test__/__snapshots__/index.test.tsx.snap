// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<Spacing
  paddingHorizontal="5px"
  paddingTop="30px"
>
  <Card
    maxWidth="480px"
    radius={true}
  >
    <Spacing
      paddingHorizontal="80px"
      paddingVertical="40px"
    >
      <LoginForm
        onLogin={[Function]}
      />
    </Spacing>
  </Card>
</Spacing>
`;
